src/components/lwc-button.ts

```diff
@@ -20,36 +24,85 @@ export class LwcButton extends withTwind(LwcElement) {
     type: Boolean,
   })
   iconOnly?: boolean;
+  @property({
+    attribute: 'connected-icon-only',
+    type: Boolean,
+  })
+  connectedIconOnly?: boolean;
 
   @property({
     type: Boolean,
   })
   disabled = false;
 
+  constructor() {
+    super();
+  }
+
+  // FIXME:
+  // - inner border icon should be a gradient
+  // - some hardcoded transparent whites should be transparent(primaryColor)
+  // TODO:
+  // - extract common button styles
+  // - extract common inner border styles
   override render() {
+    const isLoading = this._connecting || (!this._connected && this._modalOpen);
+    const iconOnly =
+      this.iconOnly || (this.connectedIconOnly && this._connected);
     return html`<div>
-      <button
-        @click=${this._onClick}
-        part="button"
-        class="${this.iconOnly ? 'p-2' : ''} ${!this.iconOnly
-          ? 'w-64 py-2 px-7'
-          : ''} font-medium font-sans shadow rounded-md flex gap-2 justify-center items-center ${this
-          .disabled
-          ? 'bg-gray-300 opacity-50'
+      <div
+        class="relative inline-flex transition-all hover:brightness-110 active:scale-95 cursor-pointer ${this
+          ._connected && !iconOnly
+          ? 'rounded-lg gap-2 justify-center items-center'
           : ''}"
-        style="${!this.disabled &&
-        'background: linear-gradient(180deg, #FFDE6E 63.72%, #F8C455 95.24%);'}"
-        ?disabled=${this.disabled}
+        style="${this._connected && !iconOnly
+          ? `background: linear-gradient(180deg, #fff6 0%, #fff0 100%), linear-gradient(180deg, ${this.colorSecondary} 0%, ${this.colorSecondary} 100%);`
```

Comment:
can this be done in a CSS class? adding a connected CSS class that then has the styles. 

using CSS variables that can be used to customize or even overwrite defaults?

Reply:
I will update to the suggested method, thanks